File: app/src/main/java/bonoj/me/athenaeum/books/BooksContract.java
package bonoj.me.athenaeum.books;

import java.util.List;

import bonoj.me.athenaeum.data.Book;

public interface BooksContract {

    interface View {

        void displayBooks(List<Book> books);

        void displayNoBooks();

        void displayError();
    }

    interface Presenter {

        //void setView(BooksContract.View view);

        void loadBooks();

        void unsubscribe();
    }
}

File: app/src/main/java/bonoj/me/athenaeum/books/BooksContract.kt
package bonoj.me.athenaeum.books

import bonoj.me.athenaeum.data.Book

interface BooksContract {

    interface View {

        fun displayBooks(books: List<Book>)

        fun displayNoBooks()

        fun displayError()
    }

    interface Presenter {

        fun loadBooks()

        fun unsubscribe()
    }
}
